third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -404,6 +403,11 @@ static ttg::AMDMfmaEncodingAttr getDotEncoding(Value inputValue,
   return getDotEncoding(user->getResult(0), opIdx);
 }
 
+// Adapted from
+// lib/Dialect/TritonGPU/Transforms/Utility.cpp::getSharedEncIfAllUsersAreDotEnc
+// to support AMDMfmaEncodingAttr.
+// TODO(max): figure out how to refactor to use upstream
+//
```

Comment:
`getSharedEncIfAllUsersAreDotEnc` could also be wholly de-duped if it weren't for the fact that we diverge in how we handle `LinearEncodingAttr` on AMD.